src/utils/ScriptTransformer.ts

```diff
@@ -110,7 +116,7 @@ export function getModuleScript(
   scripts: Map<ScriptFilePath, Script>,
 ): Script {
   let script;
-  for (const extension of validScriptExtensions) {
+  for (const extension of isLegacyScript(baseModule) ? [legacyScriptExtension] as const: scriptExtensions as ) {
```

Comment:
This is fixing a bug where it would try to import a legacy script file from an non-legacy script. The check for legacy script types is to support the ram calculation logic.